### DISCUSSION TITLE
Use of metadata variables in code chunk

### DISCUSSION THREAD

**Discussion by Pierre9344** [quarto-dev/quarto-cli#2121] [Q&A] [answered]
Hello,

In the [documentation](https://quarto.org/docs/authoring/variables.html), there is a page concerning the use of variables declared in the .yml file or in the document YAML field.
Is there a way to access those variables in the code chunk in a way similar to what the [rmarkdown knitr package does](https://bookdown.org/yihui/rmarkdown-cookbook/document-metadata.html)?

If not, is there a chance that this feature would be added later?

Thank you in advance for your answer

**Comment by dragonstyle** [comment, marked as answer]
Right now there isn't a cross language way to accomplish this, but you could use the rmarkdown syntax like

````
---
title: Test 123
---

## this is a test


```{r}
rmarkdown::metadata$title
```
````

to access document metadata and you could read the `_variables.yml` file directly in a code chunk as well...

**Reply by mcanouil** [reply]
So this is only for knitr engine.
How hard it would be to internally turn all metadata into some kind of "params" list (and equivalent for Python)? 🤔

**Reply by cscheid** [reply]
@mcanouil it's in our radar, stay tuned.

**Comment by jimjam-slam** [comment]
FWIW, I used `_variables.yml` content in [an unexecuted HTML code block](https://github.com/360-info/report-arms-trade/blob/e5d5bc75bd97c72c1b3cb9ab374eeb54680c042e/importexportmap/_popup-dialogs.qmd#L31) and [in a Javascript (_not_ OJS) block](https://github.com/360-info/report-arms-trade/blob/e5d5bc75bd97c72c1b3cb9ab374eeb54680c042e/importexportmap/_popup-buttons.qmd#L100) for a recent map we did. It did require [some HTML escaping](https://github.com/360-info/report-arms-trade/blob/main/_variables.yml) for our specific case (displaying an iframe embed code, or copying one to the clipboard), but otherwise it worked great!
